packages/server/src/routes/provider/alert/create.svelte

```diff
@@ -151,6 +152,42 @@
     }
   }
 
+  let isCustomIconSelectorOpen = false;
+  let iconSearchTerm = "";
+  let iconSearchResults = [];
+
+  function searchForCustomIcons() {
+    iconSearchResults = [];
+    const query = `{
+      search(
+        version: "5.12.1",
+        query: "${iconSearchTerm}",
+        first: 16
+      ) { id, Membership { free } }
+    }`;
+
+    fetch("https://api.fontawesome.com", {
+      method: "POST",
+      body: JSON.stringify({ query }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then(async resp => {
+        const { data } = await resp.json();
+        iconSearchResults = data.search
+          .filter(({ Membership }) => Membership.free.includes("solid"))
+          .map(icon => `fas fa-${icon.id}`);
+        if (iconSearchResults.length === 0) {
+          iconSearchResults = false;
+        }
```

Comment:
Instead of changing the type of `iconSearchResults`, let's have it be an array and do a check of both this array and the query term to determine if we have results.